server/controllers/comment.controller.js

```diff
@@ -126,6 +126,41 @@ class CommentController {
     };
     return Response(res, 201, 'Comment liked successfully', likeObject);
   }
+
+  /**
+    * @description Deletes a user's comment
+    * @param {object} req - The request object
+    * @param {object} res - The response object
+    * @returns {array} An empty array
+    */
+  static async deleteComment(req, res) {
+    const { commentId, slug } = req.params;
+    const { user: { id } } = req;
+    const articleExists = await Article.findOne({
+      where: { slug },
+    });
+    if (!articleExists) return Response(res, 404, 'Article not found.');
+
+    const articleId = articleExists.dataValues.id;
+    const table = articleExists.dataValues.published ? Comment : DraftComment;
+    const commentExists = await table.findOne({
+      where: {
+        id: commentId,
+        userId: id
+      }
+    });
+
+    if (!commentExists) return Response(res, 404, 'Comment not found.');
+
+    await table.destroy({
```

Comment:
I would like clarity for code written on line `155` and `158`

Reply:
We are deleting both the comment and all its associated thread comments. Hence the need for two calls.

Reply:
Will DELETE ON CASCADE not work here?

Reply:
@TheDrizzyWay making an extra call is quite expensive. Referencing the column on the migrations file as @Chiazokam suggested should fix this.

Reply:
There's no foreign key linking the two parties involved.